ocaml/database/dune

```diff
@@ -1,6 +1,6 @@
 (ocamllex db_filter_lex)
 
-(ocamlyacc db_filter_parse)
```

Comment:
are these wrapped around mutexes as well?

Reply:
Not directly, so either we had a race condition here on master, or we had another mutex that happened to protect this too.

We should probably backport this PR to Yangtze as well in that case, even though the Sexpression parser was safe due to the mutex, this other usage in the database (part of db_actions.ml) was not, and probably all our calls to `get_all_records_where` are not thread-safe/subject to race condition and random failures...

The xenopsd cli one doesn't matter, because that is a single-threaded CLI.